spec/system/consumer/split_checkout_spec.rb

```diff
@@ -877,6 +877,23 @@
           expect(page).to have_current_path checkout_step_path(:payment)
         end
       end
+
+      describe "order state" do
+        before do
+          visit checkout_step_path(:summary)
+        end
+
+        it "is 'confirmation' on the summary step" do
+          expect(order.reload.state).to eq "confirmation"
+        end
+
+        it "emptying the cart changes the order state back to address" do
+          visit main_app.cart_path
+          find('#clear_cart_link').click
+          expect(page).to have_current_path enterprise_shop_path(distributor)
+          expect(order.reload.state).to eq "address"
+        end
```

Comment:
Separating the specs like this is excellent style. Unfortunately, system specs are quite expensive. They take a long time to run which means that we have to wait longer and CI servers are causing more greenhouse gases etc. So in this particular case I would combine the two specs into one:
```suggestion
        it "emptying the cart changes the order state back to address" do
          visit main_app.cart_path
          expect {
            find('#clear_cart_link').click
            expect(page).to have_current_path enterprise_shop_path(distributor)
          }.to change { order.reload.state }.from("confirmation").to("address")
        end
```

Reply:
Thanks @mkllnk ,

> `to change { order.reload.state }.from("confirmation").to("address")`

this formulation is really useful to test state changes :+1: